tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -29,41 +29,59 @@ class AgentTestSuitesCombinatorSchema(schema.Combinator):
     cloud: str = field(
         default_factory=str, metadata=field_metadata(required=True)
     )
-    image: str = field(
-        default_factory=str, metadata=field_metadata(required=True)
-    )
     location: str = field(
         default_factory=str, metadata=field_metadata(required=True)
     )
+    image: str = field(
+        default_factory=str, metadata=field_metadata(required=False)
+    )
     vm_size: str = field(
-        default_factory=str, metadata=field_metadata(required=True)
+        default_factory=str, metadata=field_metadata(required=False)
+    )
+    vm_name: str = field(
+        default_factory=str, metadata=field_metadata(required=False)
     )
 
 
 class AgentTestSuitesCombinator(Combinator):
     """
-    The "agent_test_suites" combinator returns a list of items containing six variables that specify the environments
-    that the agent test suites must be executed on:
+    The "agent_test_suites" combinator returns a list of variables that specify the environments (i.e. test VMs) that the agent
+    test suites must be executed on:
 
+        * c_env_name: Unique name for the environment, e.g. "0001-com-ubuntu-server-focal-20_04-lts-westus2"
         * c_marketplace_image: e.g. "Canonical UbuntuServer 18.04-LTS latest",
         * c_location: e.g. "westus2",
         * c_vm_size: e.g. "Standard_D2pls_v5"
         * c_vhd: e.g "https://rhel.blob.core.windows.net/images/RHEL_8_Standard-8.3.202006170423.vhd?se=..."
         * c_test_suites: e.g. [AgentBvt, FastTrack]
-        * c_name: Unique name for the image, e.g. "0001-com-ubuntu-server-focal-20_04-lts-westus2"
 
     (c_marketplace_image, c_location, c_vm_size) and vhd are mutually exclusive and define the environment (i.e. the test VM)
     in which the test will be executed. c_test_suites defines the test suites that should be executed in that
     environment.
+
+    The 'vm_name' runbook parameter can be used to execute the test suites on an existing VM. In that case, the combinator
+    generates a single item with these variables:
+
+        * c_env_name: Name for the environment, same as vm_name
+        * c_vm_name:  Name of the test VM
+        * c_location: Location of the test VM e.g. "westus2",
+        * c_test_suites: e.g. [AgentBvt, FastTrack]
     """
     def __init__(self, runbook: AgentTestSuitesCombinatorSchema) -> None:
         super().__init__(runbook)
-        self._environments = self.create_environment_list()
-        self._index = 0
-
         if self.runbook.cloud not in self._DEFAULT_LOCATIONS:
             raise Exception(f"Invalid cloud: {self.runbook.cloud}")
 
+        if self.runbook.vm_name != '' and (self.runbook.image != '' or self.runbook.vm_size != ''):
```

Comment:
The combinator was generating the full list of distros when running on an existing VM. For that scenario, it should generate a single item containing that VM